packages/ui/src/components/TalkList/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+import { ListTalk } from "types";
+
+import { voteTalk } from "../../api";
+
+import { VoteIcon } from "../VoteIcon";
+import { ResourceLoader } from "../ResourceLoader";
+
+import styles from "./styles.module.css";
+
+export interface TalkListProps {
+  start: number;
+  talks: ListTalk[];
+}
+
+export const TalkList: React.FC<TalkListProps> = (props) => {
+  const { start, talks } = props;
+
+  const [sorted, setSorted] = useState(talks);
+
+  return (
+    <>
+      {sorted.map((talk, i) => (
+        <Item
+          key={talk.id}
+          index={start + i}
+          talk={talk}
+          onVoted={(votes) => {
+            if (talk.votes === votes) return;
+            talk.votes = votes;
+            const nextSorted = sorted.slice();
+            nextSorted.sort((a, b) => (a.votes > b.votes ? -1 : 1));
+            setSorted(nextSorted);
+          }}
+        />
+      ))}
+    </>
+  );
+};
+
+interface ItemProps {
+  index: number;
+  talk: ListTalk;
+  onVoted: (votes: number) => void;
+}
+
+const Item: React.FC<ItemProps> = ({ index, talk, onVoted }) => {
+  const [voting, setVoting] = useState<Promise<{ votes: number }>>(Promise.resolve(talk));
```

Comment:
What's the benefit of using `Promise.resolve(talk)` and `ResourceLoader` here?

Reply:
I would like the votes number to be responsive. That means, when people vote, they should see a loading instead of the `votes: {number}`. But when there is no voting request, it should show initial votes. So I made the initial value to be the default of the voting promise.